test/tests/functional/pbs_multi_sched.py

```diff
@@ -580,6 +580,60 @@ def test_run_limts_per_scheduler(self):
         jc = "Not Running: User has reached server running job limit."
         self.server.expect(JOB, {'comment': jc}, id=jid4)
 
+    def test_list_multi_sched(self):
+        """
+        Test to verify that qmgr list sched works when multiple schedulers are present
+        """
+
+        self.setup_sc1()
+        self.setup_sc2()
+        self.setup_sc3()
+
+        self.server.manager(MGR_CMD_LIST, SCHED, expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="default", expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc1", expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc2", expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc3", expect=True)
+
+        try:
+            self.server.manager(MGR_CMD_LIST, SCHED, id="invalid_scname")
+        except PbsManagerError as e:
+            err_msg = "Unknown Scheduler"
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
+        # delete sc3 sched
+        self.server.manager(MGR_CMD_DELETE, SCHED, id="sc3", sudo=True)
+
+        try:
+            self.server.manager(MGR_CMD_LIST, SCHED, id="sc3")
+        except PbsManagerError as e:
+            err_msg = "Unknown Scheduler"
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
+        self.server.manager(MGR_CMD_LIST, SCHED, expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="default", expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc1", expect=True)
+
+        self.server.manager(MGR_CMD_LIST, SCHED, id="sc2", expect=True)
+
+        # delete sc1 sched
+        self.server.manager(MGR_CMD_DELETE, SCHED, id="sc1", sudo=True)
+
+        try:
+            self.server.manager(MGR_CMD_LIST, SCHED, id="sc1")
+        except PbsManagerError as e:
+            err_msg = "Unknown Scheduler"
+            self.assertTrue(err_msg in e.msg[0],
+                            "Error message is not expected")
+
```

Comment:
One more thing that you might wanna test is calling manager() list with id = None, that will cause it to stat all scheduler objects.

Reply:
Its already done here 
https://github.com/Shrini-h/pbspro/blob/e27d7ffa81f8fc6c44d5a54b52a04bd5f1bd475f/test/tests/functional/pbs_multi_sched.py#L592
and here 
https://github.com/Shrini-h/pbspro/blob/e27d7ffa81f8fc6c44d5a54b52a04bd5f1bd475f/test/tests/functional/pbs_multi_sched.py#L613

Reply:
I see, that's cool. But I think that you also need to actually check that the manager call did the right thing. A manager list call updated the respective scheduler object's 'attribute' dictionary. I think the test should check that the dictionaries got populated correctly, at least for one manager call.

Reply:
Added lines to test case to check updation of attributes dictionary of sched sc2 from list sched command